GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -274,6 +274,17 @@ public override bool TryDownloadNewestVersion(out string errorMessage)
                 }
             }
 
+            if (!this.noVerify)
+            {
+                if (!this.nuGetFeed.VerifyPackage(this.DownloadedPackagePath, this.nuGetUpgraderConfig.CertificateFingerprint))

```

Comment:
Add a mock for `VerifyPackage` to `private Mock<NuGetFeed> mockNuGetFeed` in `NuGetUpgraderTests` and you should be able to add a unit test the `TryDownloadNewestVersion` fails if package verification fails.

(If there are other unit tests code paths that hit `VerifyPackage` they can be updated as well)